src/core/crypto_tls.c

```diff
@@ -664,6 +656,32 @@ QuicCryptoTlsReadInitial(
     return QUIC_STATUS_SUCCESS;
 }
 
+_IRQL_requires_max_(PASSIVE_LEVEL)
+QUIC_STATUS
+QuicCryptoTlsReadClientRandom(
+    _In_ QUIC_CONNECTION* Connection,
+    _In_reads_(BufferLength)
+        const uint8_t* Buffer,
+    _In_ uint32_t BufferLength,
+    _Inout_ QUIC_TLS_SECRETS* TlsSecrets
+    )
+{
+    if (BufferLength < TLS_MESSAGE_HEADER_LENGTH + sizeof(uint16_t) + TLS_RANDOM_LENGTH) {
```

Comment:
Can't you assert this instead? It was already validated with a previous call.

Reply:
It'll generate a SAL issue without a check